Project-1/README.md

```diff
@@ -63,60 +68,256 @@ T	T	F	T	T
+
+used CNF calculator https://www.artima.com/cs/cnf.html & https://www.erpelstolz.at/gateway/formular-uk-zentral.html
+
 
 3. Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+
   ```(a)  B(x,y) that says that x is a brother of y
   (b)  A(x,y) that says that x is an aunt of y
   (c)  C(x,y) that says that x and y are cousins 
   (d)  O(x) that says that x is an only child
   (e)  T(x) that says that x has exactly two brothers 
-```
+
+```(a) ∃x∃y∃z(P(z,y)∧P(z,x)∧(x≠y)∧(z≠x)∧(z≠y)∧¬Fx) - There is some x and some y and some z such that z is the parent of y and z is the parent x and x is not female
+(a) A brother is an individual who identifies as male and has at least one sibling that is, they share at least one parent. Half-siblings and step-siblings can be brothers. In the example above, x would be an individual who shares at least one parent with y and x would identify as male.
+
+(b) ∃w∃x∃y∃z(P(z,y)∧P(w,z)∧P(w,x)∧Fx) - There is some w and some x and some y and some z such that z is the parent of y and w is the parent of z and w is the parent of x and x is female
+(b) An Aunt is a female identifying sibling of a parent. In the example above x would be an Aunt of y so long as x is the child of y's sibling
+
+(c) ∃v∃w∃x∃y∃z(P(v,w)∧P(v,z)∧P(w,x)∧P(z,y)) - There is some v and some w and some x and some y and some z such that v is the parent of w and v is the parent of z and w is the parent of x and z is the parent of y
+(c) A cousin is a relationship between two individuals whose parents are siblings. Such that you have a cousin in the child of your aunt
+
+(d) ∃x∃y∀z(P(y,z)→z=x) - There is some x and some y such that for all z if y is the parent of z then z and x are identical 
+(d) An only child is an individual who was raised in a house primarily without siblings. Individuals who have step or half siblings but were raised pimarily alone are only children. 
+
+(e) ∃v∃w∃x∃y∀z(P(v,x)∧P(v,w)∧¬Fw∧P(v,y)∧¬Fy∧∀z((P(v,z)∧¬Fz)→((z=x)∨(z=w)∨(z=y)))) - There is some v and some w and some x and some y such that for all z if v is the parent of x and v is the parent of w and w is not female and v is the parent of y and y is not female and for all instances of z v is a parent of z and z is not female ten z is identical with w or z is identical with w or z is identical with y
+(e) An individual has exactly two brothers if and only if they have two siblings who identify as male.
 
 4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   ```(a)  B that says that x is a brother of y
```

Comment:
I see that you have the answers for these on 5, but after talking with John, he told @giacomodecolle and I that using the inverse of parent_of is not possible to define since you need a direct definition of parent_of. @peihongx this is for you as well.
But if you assume parent_of, you can define inverse terms that use parent_of indirectly. 
for example, Giacomo and I defined P2 (parent of 2) := (∃parent_of. ≥ 2) ⊓ ¬(∃parent_of. = 1) and then defined it's inverse. This is fine since we are not using the definition of parent_of directly for the inverse. So, this is what we got for part a:
(a)  B that says that x is a brother of y

    person := M ⊔ ¬M

    P2 (parent of 2) := (∃parent_of. ≥ 2) ⊓ ¬(∃parent_of. = 1)

    B (brother) := M ⊓ ∃P2¯.person

Also, you need to remember the difference between RoleNames and ConceptNames.

Reply:
> I see that you have the answers for these on 5, but after talking with John, he told @giacomodecolle and I that using the inverse of parent_of is not possible to define since you need a direct definition of parent_of. @peihongx this is for you as well. But if you assume parent_of, you can define inverse terms that use parent_of indirectly. for example, Giacomo and I defined P2 (parent of 2) := (∃parent_of. ≥ 2) ⊓ ¬(∃parent_of. = 1) and then defined it's inverse. This is fine since we are not using the definition of parent_of directly for the inverse. So, this is what we got for part a: (a) B that says that x is a brother of y
> 
> ```
> person := M ⊔ ¬M
> 
> P2 (parent of 2) := (∃parent_of. ≥ 2) ⊓ ¬(∃parent_of. = 1)
> 
> B (brother) := M ⊓ ∃P2¯.person
> ```
> 
> Also, you need to remember the difference between RoleNames and ConceptNames.

@ahasanzadeh13 Happy for you'all's new progress on this! Interesting

Reply:
@ahasanzadeh13 @giacomodecolle This is the proposed definition of Aunt (x is an aunt of y):
A := ¬M ⊓ (∃p2¯.((parent_of. ≥ 2).(parent_of.person)) ⊔ ¬parent_of.person)

In my view, the best way of writing this formula seems to be as follows:

A ≡ ¬M ⊓ (∃p2¯.(≥ 2 parent_of. (∃parent_of. Person)) ⊔ ¬∃parent_of. Person)

The reason is, when we try to use both Concept (Name) and Role (Name) in a DL-formula, it is either (≥ n r. C) or (≤ n r. C) or (∃r. C) or (∀r. C). That is to mean, a quantifier must be placed immediately before a role name unless the role name is restricted by a numerical. Also, ":=" seems not to be a symbol in DL, but "≡" is.

I think similar syntactic changes may also be applied to other definitions in Question 4.

Ok, the above is a comment on syntax. But another worry is about semantics. This definition of Aunt seems to say: A guy x is an Aunt iff:
(i) x is female, **and** 
(ii) [**(a)** x is one of at least two kids produced by a parent w, who (w) has at least two kids _each of who_ has a kid; **or** **(b)** x has no kid.]

When we say in (a) that w has at least two kids, w may have 5 kids! In that possible case, (a) implies that each of those 5 kids of w has a kid. But this seems too strong because it is enough that only one of those 5 kids (say z) has a kid who happens to be y. Only adding (b) does not exercise a reasonable limit here.

Reply:
@peihongx, I read through all that you wrote and the way you are reasoning seems right to me! I am going to adjust my answers and if Professor Beverley does not mind having multiple possible answers for #4 (fingers crossed) then I will have a few possible answers ready. 

Great work, Karl. It seems you found alternative means to answering these questions. Thanks for sharing your process!